src/vs/workbench/api/common/extHostTreeViews.ts

```diff
@@ -406,6 +415,16 @@ class ExtHostTreeView<T> extends Disposable {
 		this.proxy.$setTitle(this.viewId, this._title, description);
 	}
 
+	private _badge: vscode.ViewBadge | undefined;
+	get badge(): vscode.ViewBadge | undefined {
+		return this._badge;
+	}
+
+	set badge(badge: vscode.ViewBadge | undefined) {
```

Comment:
We probably should skip the update here if the existing badge matches the current badge. This would be helpful in case a extension ends up mistakenly trying to update the badge multiple times in a loop

Reply:
Done.